### PR TITLE
Added: New order item argument for commission calculation for 1085

### PR DESCRIPTION
**Pull request by mahlamusa**
### All Submissions:

* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Added a new argument to pass the order item from which commissions is calculated. This will allow commission to be calculated from the item instead of product price.

Required by https://github.com/wcvendors/wcvendors-pro/pull/1085.

### Testing

1. Checkout this branch and test https://github.com/wcvendors/wcvendors-pro/pull/1085

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?

### CODE REVIEW

**Comment by digitalchild** [comment]
This looks good, @dinhtungdu please confirm and we'll get this merged and push another release for wc vendors marketplace.

File: classes/class-commission.php [hunk 1]
```diff
@@ -335,18 +335,27 @@ public static function get_commission_rate( $product_id ) {
...
+	 * @param WC_Order $order The WooCommerce order.
+	 * @param int      $qty The product quantity.
+	 * @param mixed    $item The order item. Optional, default array().
 	 * @return float
 	 */
-	public static function calculate_commission( $product_price, $product_id, $order, $qty ) {
+	public static function calculate_commission( $product_price, $product_id, $order, $qty, $item = array() ) {
+
+		if ( is_a( $item, 'WC_Order_Item_Product' ) ) {
+			$product_price = $item->get_total();
```
Comment (dinhtungdu) [inline classes/class-commission.php]: Here we need to care about the coupon setting to get the correct product price:
- `$item->get_total()`: after coupon
- `$item->get_sub_total()`: before coupon

File: classes/admin/class-vendor-reports.php [hunk 2]
```diff
@@ -87,7 +89,7 @@ public function filter_products( $orders ) {
 				continue;
 			} else {
 				if ( ! empty( $order->line_total ) ) {
-					$orders[ $key ]->line_total = WCV_Commission::calculate_commission( $order->line_total, $order->product_id, $order, $order->qty );
+					$orders[ $key ]->line_total = WCV_Commission::calculate_commission( $order->line_total, $order->product_id, $order, $order->qty, $item );
```
[the remark this answers is not in the document]
Comment (mahlamusa) [reply classes/admin/class-vendor-reports.php]: Removed $item not used in this line

File: classes/class-commission.php [hunk 3]
```diff
@@ -335,18 +335,27 @@ public static function get_commission_rate( $product_id ) {
...
-	 * @param float   $product_price
-	 * @param unknown $product_id
-	 *
+	 * @version 2.1.14
+	 * @since   2.0.0
+	 * @param float    $product_price The product price.
+	 * @param int      $product_id The product id.
+	 * @param WC_Order $order The WooCommerce order.
+	 * @param int      $qty The product quantity.
+	 * @param mixed    $item The order item. Optional, default array().
```
[the remark this answers is not in the document]
Comment (mahlamusa) [reply classes/class-commission.php]: Now aligned

File: classes/class-commission.php (hunk 1 above)
Comment (mahlamusa) [reply classes/class-commission.php]: Added a filter to be used for the commission_coupon_action, the setting is on Pro, now Pro will filter he commission action based on its settings, see commit: https://github.com/wcvendors/wcvendors-pro/pull/1085/commits/42c2b490a4c1d175d6ff478037c3d46d6c367114.

Comment (digitalchild) [reply classes/class-commission.php]: This should not be done in wc vendors marketplace. Marketplace does not support coupons and this code should not be in here. This should only parse the item through to the filter for use by other integrations including pro.

Comment (mahlamusa) [reply classes/class-commission.php]: Removed coupon related code.

File: classes/admin/class-vendor-reports.php [hunk 4]
```diff
@@ -67,7 +67,9 @@ public function filter_tabs( $tabs ) {
 	/**
 	 * Filter products based on current vendor
 	 *
-	 * @param unknown $orders
+	 * @version 2.1.14
+	 * @since   2.0.0
+	 * @param unknown $orders List of orders.
```
Comment (dinhtungdu) [inline classes/admin/class-vendor-reports.php]: Why is this unknown?

**Comment by digitalchild** [comment]
@dinhtungdu vendor reports are deprecated so they are never actually used. This can be ignored.

**Comment by digitalchild** [comment]
LGTM!

**Pull request merged**